### PR TITLE
DatePicker: fix unknown prop warning

### PR DESCRIPTION
**Pull request by Copons**
Updated `react-day-picker` to last version and added a `locale.formatDay()` to avoid a fatal error in the external component.

See: https://github.com/Automattic/wp-calypso/issues/7413

Test live: https://calypso.live/?branch=fix/date-picker-props-warnings

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -92,7 +92,7 @@
     "react-addons-shallow-compare": "15.3.0",
     "react-addons-update": "15.3.0",
     "react-click-outside": "2.1.0",
-    "react-day-picker": "1.3.2",
+    "react-day-picker": "2.4.1",
```
Comment (retrofox) [inline package.json]: I think you should _shrinkwrapy_ to lock down our dependency versions.
Doc page -> https://github.com/Automattic/wp-calypso/blob/master/docs/shrinkwrap.md

Comment (Copons) [inline package.json]: I've never used shrinkwrap, I'm gonna look into it asap.

Comment (retrofox) [inline package.json]: In short you should run the command in the CLI immediately after you update some dependency:

```cli
> make shrinkwarp
```

Thus the `npm-shrinkwrap.json` will be update. Finally just add and commit the changes of this file. But yes, take a look to doc is a good idea.

Comment (Copons) [inline package.json]: @retrofox FIY I've run `make shrinkwrap` and it updated a bunch of dependencies that I guess are related to `react-day-picker`.
After cleaning and reinstalling, (to my knowledge) Calypso seems to work as expected.

Comment (retrofox) [inline package.json]: Testing again now.

**Comment by retrofox** [comment]
LGTM/ 🚢 :-D

**Comment by retrofox** [comment]
ok @Copons your patch has been deployed to `prod`! thanks.

**Pull request merged**